win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1577,16 +1593,34 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
     PVOID Buffer = NULL;
     IO_STATUS_BLOCK Iosb;
     PVOID SectionObject;
-    SIZE_T ViewSize = 0;
+    SIZE_T ViewSize = 0, Length;
     LARGE_INTEGER SectionSize;
     OBJECT_ATTRIBUTES ObjectAttributes;
     GDI_LOAD_FONT LoadFont;
     INT FontCount;
     HANDLE KeyHandle;
+    UNICODE_STRING PathName;
+    WCHAR szPath[MAX_PATH];
     static const UNICODE_STRING TrueTypePostfix = RTL_CONSTANT_STRING(L" (TrueType)");
 
+    /* Build PathName */
+    Length = wcslen(SharedUserData->NtSystemRoot);
+    FileName->Buffer[FileName->Length / sizeof(WCHAR)] = 0;
+    if (RtlCompareMemory(FileName->Buffer, SharedUserData->NtSystemRoot,
+                         Length * sizeof(WCHAR)) == Length * sizeof(WCHAR))
+    {
+        RtlStringCbCopyW(szPath, sizeof(szPath), L"\\SystemRoot");
+        RtlStringCbCatW(szPath, sizeof(szPath), &FileName->Buffer[Length]);
```

Comment:
Are we sure there the filename always start with a backslash? If not you need to append '\\' (also known as OBJ_PATH_SEPARATOR) before concatenating the filename.
Also since the filename is a counted string you need to use RtlStringCbCatNW instead.

Reply:
Yes, we are sure.